src/utils/details.js

```diff
@@ -84,9 +90,26 @@ function parseDetails ({
   secret
 }) {
   const detailsBuffer = Buffer.from(details, 'base64')
-  const publicRequest = _parseRequest(detailsBuffer)
-  const decrypted = cryptoHelper.aesDecryptBuffer(publicRequest.data, secret)
-  const privateRequest = _parseRequest(decrypted)
+  const publicRequest = _parseRequest({
+    request: detailsBuffer,
+    statusLine: true
+  })
+
+  const [ , token ] = publicRequest.headers['key'].match(KEY_HEADER_REGEX)
```

Comment:
I think if this doesn't match, this line will throw `TypeError: Cannot read property 'Symbol(Symbol.iterator)' of null`